exonum/src/blockchain/service.rs

```diff
@@ -408,6 +409,32 @@ impl SharedNodeState {
                 .peers_info
                 .insert(c.addr(), *p);
         }
+
+        self.update_majority_count(state);
+    }
+
+    /// Returns the majority count from the current "State"
+    /// of a blockchain node.
+    pub fn majority_count(&self) -> usize {
+        self.state
+            .read()
+            .expect("Expected read lock.")
+            .majority_count
+    }
+
+    /// Updates internal majority count, from `State` of a blockchain node.
+    fn update_majority_count(&self, state: &State) {
+        let mut majority_count = state.majority_count();
+
+        let peers: Vec<SocketAddr> = self.peers_info().iter().map(|(s, _k)| (*s)).collect();
```

Comment:
For some reason previous comment is marked as outdated, but once again:
I think it'll be better to not use `collect` here. 
Something like this:
```
        let peers = self.peers_info();
        if peers
            .iter()
            .map(|(s, _k)| (*s))
            .find(|x| x == &state.our_connect_message().addr())
            .is_none()
        {
            majority_count -= 1;
        }
```

And an explaining comment, why are we decrementing majority count.

Reply:
It will be removed at all in next commit